dm/master/openapi.go

```diff
@@ -133,17 +134,100 @@ func (s *Server) DMAPIDeleteSource(ctx echo.Context, sourceName string) error {
 
 // DMAPIStartRelay url is:(POST /api/v1/sources/{source-id}/relay).
 func (s *Server) DMAPIStartRelay(ctx echo.Context, sourceName string) error {
-	return nil
+	var req openapi.StartRelayRequest
+	if err := ctx.Bind(&req); err != nil {
+		return err
+	}
+	sourceCfg := s.scheduler.GetSourceCfgByID(sourceName)
+	if sourceCfg == nil {
+		return terror.ErrSchedulerSourceCfgNotExist.Generate(sourceName)
+	}
+	// update relay related in source cfg
+	sourceCfg.EnableRelay = true
+	if req.RelayBinlogName != nil && sourceCfg.RelayBinLogName != *req.RelayBinlogName {
+		sourceCfg.RelayBinLogName = *req.RelayBinlogName
+	}
+	if req.RelayBinlogGtid != nil && sourceCfg.RelayBinlogGTID != *req.RelayBinlogGtid {
+		sourceCfg.RelayBinlogGTID = *req.RelayBinlogGtid
+	}
+	if req.RelayDir != nil && sourceCfg.RelayDir != *req.RelayDir {
+		sourceCfg.RelayDir = *req.RelayDir
+	}
+	if purge := req.Purge; purge != nil {
+		if purge.Expires != nil && sourceCfg.Purge.Expires != *purge.Expires {
+			sourceCfg.Purge.Expires = *purge.Expires
+		}
+		if purge.Interval != nil && sourceCfg.Purge.Interval != *purge.Interval {
+			sourceCfg.Purge.Interval = *purge.Interval
+		}
+		if purge.RemainSpace != nil && sourceCfg.Purge.RemainSpace != *purge.RemainSpace {
+			sourceCfg.Purge.RemainSpace = *purge.RemainSpace
+		}
+	}
+	// update current source relay config before start relay
+	if err := s.scheduler.UpdateSourceCfg(sourceCfg); err != nil {
+		return err
+	}
+	return s.scheduler.StartRelay(sourceName, []string{req.WorkerName})
```

Comment:
will we support batch start-relay for workers?

Reply:
it depends on needs, maybe we can add a batch-start-relay api in source api